### ISSUE TITLE
mrcalc: Erroneous stack formation on dev

### ISSUE THREAD

**Issue by Lestropie** [MRtrix3/mrtrix3#3017]
```ShellSession
robertes@9520l-003953-l:~$ ~/src/3.0.4/bin/mrcalc pi
3.14159
robertes@9520l-003953-l:~$ ~/src/mrtrix3/build_release/bin/mrcalc pi
3.14159
```

All good thus far...

```ShellSession
robertes@9520l-003953-l:~$ ~/src/3.0.4/bin/mrcalc pi 4 -mult
12.5664
robertes@9520l-003953-l:~$ ~/src/mrtrix3/build_release/bin/mrcalc pi 4 -mult
3.14159
robertes@9520l-003953-l:~$ ~/src/mrtrix3/build_release/bin/mrcalc 4 pi -mult
4
```

:grimacing: 

Best guess is it's related to #2911?

**Comment by daljit46** [comment]
Yes, it was indeed due to a change in #2911. It's probably worth adding some tests for `mrcalc`'s pure calculator mode. Will make a PR later.

**Comment by daljit46** [comment]
Closed by #3019
